File: 2022/2022W19/preppindataW19.R
library(readxl)
library(dplyr)

xlsx <- "C:/Data/PreppinData/PD 2022 Wk 19 Input.xlsx"

sales <- read_excel(xlsx, 'Sales') %>%
  merge(read_excel(xlsx, 'Size Table') %>% rename('Sales Size' = 'Size'), by.x='Size', by.y='Size ID') %>%
  merge(read_excel(xlsx, 'Product Set') %>%
          rename('Product Size' = 'Size') %>%
          mutate('Product' = gsub('S','',`Product Code`)), by='Product')
  
correct_sales <- sales %>%
  filter(`Sales Size`==`Product Size`) %>%
  select(c('Product Size', 'Scent', 'Product', 'Store'))

wrong_sales <- sales %>%
  filter(`Sales Size`!=`Product Size`) %>%
  group_by(`Product Code`, `Product Size`, `Scent`) %>%
  summarise('Number of Sales with wrong size' = n(), .groups='drop')
  
View(correct_sales)
View(wrong_sales)

File: 2022/2022W19/PreppindataW19.py
import pandas as pd

xlsx = pd.ExcelFile(r"C:\Data\PreppinData\PD 2022 Wk 19 Input.xlsx")

prodset = pd.read_excel(xlsx, 'Product Set').rename(columns={'Size':'Product Size'})
prodset['Product Code'] = prodset['Product Code'].str.replace('S','')

sales = pd.read_excel(xlsx, 'Sales')
size = pd.read_excel(xlsx, 'Size Table').rename(columns={'Size':'Sales Size'})

sales = sales.merge(size, left_on='Size', right_on='Size ID')
sales = sales.merge(prodset, left_on='Product', right_on='Product Code')

correct_sales = sales.query('`Sales Size`==`Product Size`')[['Product Size', 'Scent', 'Product', 'Store']]

wrong_sales = sales.query('`Sales Size`!=`Product Size`')[['Product Size', 'Scent', 'Product Code', 'Store']]
wrong_sales = wrong_sales.groupby(['Product Code', 'Product Size', 'Scent'], as_index=False).agg({'Store':'count'})
wrong_sales = wrong_sales.rename(columns={'Store':'Number of Sales with wrong size'})
